protocol/chainlib/chain_fetcher.go

```diff
@@ -84,10 +84,15 @@ func (cf *ChainFetcher) Validate(ctx context.Context) error {
 
 func (cf *ChainFetcher) populateCache(relayData *pairingtypes.RelayPrivateData, reply *pairingtypes.RelayReply, requestedBlockHash []byte, finalized bool) {
 	if requestedBlockHash != nil || finalized {
+		relayPrivateDataBytes, marshalErr := relayData.Marshal()
```

Comment:
let's create a copy function with marshal unmarshal that returns the copy or an error, this will be prettier and less changes in files